packages/rpc-subscriptions-spec/src/__tests__/rpc-subscriptions-pubsub-plan-test.ts

```diff
@@ -0,0 +1,316 @@
+            signal: abortController.signal,
+            subscribeMethodName: 'thingSubscribe',
+            subscribeParams: [],
+            unsubscribeMethodName: 'thingUnsubscribe',
+        });
+        expect(mockChannel.on).toHaveBeenCalledWith('error', expect.any(Function), {
+            signal: abortController.signal,
+        });
+    });
+    it('sends the expected subscribe message', () => {
+        const expectedParams = [1, 2, 3];
+        executeRpcPubSubSubscriptionPlan({
+            channel: mockChannel as RpcSubscriptionsChannel<unknown, unknown>,
+            signal: abortController.signal,
+            subscribeMethodName: 'thingSubscribe',
+            subscribeParams: expectedParams,
+            unsubscribeMethodName: 'thingUnsubscribe',
+        });
+        expect(mockSend).toHaveBeenCalledWith(
+            expect.objectContaining({
+                id: expect.any(Number),
+                jsonrpc: '2.0',
+                method: 'thingSubscribe',
+                params: expectedParams,
+            }),
+        );
+    });
+    describe('given that the subscribe message fails to send', () => {
+        beforeEach(() => {
+            mockSend.mockRejectedValue('o no');
+        });
+        it("rejects with the send method's rejection", async () => {
+            expect.assertions(1);
+            const publisherPromise = executeRpcPubSubSubscriptionPlan({
+                channel: mockChannel as RpcSubscriptionsChannel<unknown, unknown>,
+                signal: abortController.signal,
+                subscribeMethodName: 'thingSubscribe',
+                subscribeParams: [],
+                unsubscribeMethodName: 'thingUnsubscribe',
+            });
+            await expect(publisherPromise).rejects.toBe('o no');
+        });
+        it('does not send an unsubscribe message when aborted', () => {
+            expect.assertions(1);
+            mockSend.mockClear();
+            abortController.abort();
+            expect(mockSend).not.toHaveBeenCalled();
+        });
+    });
+    describe('given that the server has not yet acknowledged the subscription', () => {
+        let publisherPromise: ReturnType<typeof executeRpcPubSubSubscriptionPlan>;
+        beforeEach(() => {
+            publisherPromise = executeRpcPubSubSubscriptionPlan({
+                channel: mockChannel as RpcSubscriptionsChannel<unknown, unknown>,
+                signal: abortController.signal,
+                subscribeMethodName: 'thingSubscribe',
+                subscribeParams: [],
+                unsubscribeMethodName: 'thingUnsubscribe',
+            });
+        });
+        afterEach(() => {
+            publisherPromise.catch(() => {});
+        });
+        it('rejects when aborted', async () => {
+            expect.assertions(1);
+            abortController.abort();
+            await expect(publisherPromise).rejects.toThrow();
+        });
```

Comment:
I just want to point out, @lorisleiva, that this is a material change from the status quo, in that this ensures that an abort event nukes the subscription with an `AbortError` in the event that it's aborted before the subscription gets set up.